### PR TITLE
[SYCL][E2E] Fix bindless images tests to run on Level Zero devices

### PR DESCRIPTION
**Pull request by nrspruit**

### CODE REVIEW

**Comment by alycm** [comment]
I don't know if it important, but the test configuration changes that were previously reverted which more constrained than running on all level zero devices: https://github.com/intel/llvm/pull/14245

**Comment by nrspruit** [comment]
Currently, all the bindless support added to L0 is not being validated because the tests excluded all but CUDA.

File: sycl/test-e2e/bindless_images/read_1D.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// REQUIRES: cuda
+// REQUIRES: cuda || (level_zero && gpu-intel-dg2)
 
 // RUN: %{build} -o %t.out
 // RUN: %t.out
```
Comment (wenju-he) [inline sycl/test-e2e/bindless_images/read_1D.cpp:4]:
```suggestion
// RUN: env NEOReadDebugKeys=1 UseBindlessMode=1 UseExternalAllocatorForSshAndDsh=1 %t.out
```

File: sycl/test-e2e/bindless_images/read_1D.cpp [hunk 2]
```diff
@@ -1,7 +1,7 @@
-// REQUIRES: cuda
+// REQUIRES: cuda || (level_zero && gpu-intel-dg2)
 
 // RUN: %{build} -o %t.out
-// RUN: %t.out
+// RUN: env NEOReadDebugKeys=1 UseBindlessMode=1 UseExternalAllocatorForSshAndDsh=1 %t.out
```
Comment (ProGTX) [inline sycl/test-e2e/bindless_images/read_1D.cpp:4]: How come only this test requires these environment variables? Maybe we should add them to the LIT environment for all Bindless tests?

File: sycl/test-e2e/bindless_images/read_1D.cpp [hunk 3]
```diff
@@ -1,7 +1,7 @@
-// REQUIRES: cuda
+// REQUIRES: cuda || (level_zero && gpu-intel-dg2)
```
Comment (ProGTX) [inline sycl/test-e2e/bindless_images/read_1D.cpp:1]: Doesn't this require a specific driver version? Should that be documented somewhere?

File: sycl/test-e2e/bindless_images/read_1D.cpp (hunk 2 above)
Comment (wenju-he) [reply sycl/test-e2e/bindless_images/read_1D.cpp:4]: these env are required for all tests in this PR

**Comment by wenju-he** [comment]
This PR could be submitted once 24.26 gpu driver is uplifted in https://github.com/intel/llvm/pull/14838

In addition, array/read_write_unsampled_array.cpp is also passing with 24.26 driver. So this test can also be enabled in this PR.

**Comment by nrspruit** [comment]
@wenju-he , the PR for the gpu uplift was completed, please check to see if there are any other issues such that we have the opportunity to merge.

**Review by ProGTX — APPROVED** [review]
Specifying the environment in each test individually is not the preferred solution, but I think it's best we get this in ASAP.

**Review by wenju-he — APPROVED** [review]
LGTM

My local testing shows 17 tests pass.
Note vulkan_interop/unsampled_images.cpp is unsupported:
```
UNSUPPORTED: SYCL :: bindless_images/vulkan_interop/unsampled_images.cpp (78 of 95)
No supported devices to run the test on
```

**Comment by wenju-he** [comment]
@nrspruit  vulkan_interop/unsampled_images.cpp requires import_external_semaphore support, while vulkan_interop/sampled_images.cpp doesn't.
Please double check if PR should enable vulkan_interop/sampled_images.cpp instead.

**Comment by wenju-he** [comment]
the new commit LGTM

**Comment by nrspruit** [comment]
@intel/llvm-gatekeepers please merge, thank you

**Pull request merged**